client/src/components/Avatar.js

```diff
@@ -19,39 +20,79 @@ const UploadImage = styled.button`
   width: 25px;
   margin-left: 35px;
   position: absolute;
+  align-items: center;
   justify-content: center;
+  background: url(${cameraIcon}) no-repeat;
+  background-position: center;
   border-radius: 50%;
   border: none;
   background-color: white;
   outline: none;
+  cursor: pointer;
   &:active {
     transform: translateY(1px) scale(0.9);
     box-shadow: 0px 5px 10px rgba(0, 0, 0, 20%);
   }
 `;
 
+const ImageInput = styled.input`
+  display: flex;
+  height: 20px;
+  width: 20px;
+  opacity: 0%;
+   position relative;
+   border-radius: 50%;
+  border: none;
+  focus {
+    outline: none;
+  }
+  overflow: hidden;
+`;
+
 const AvatarContainer = styled.div`
   display: flex;
-  width: 140px;
   height: auto;
   flex-direction: row-reverse;
   justify-content: center;
   align-items: flex-end;
 `;
-const Avatar = ({ onClick, src }) => {
+const Avatar = () => {
+  const { image, setImage } = useAvatar();
```

Comment:
In this pull request, I don't see the advantage of a context provider. Do you want to update the avatar in a different component the same time you change it?
Right now, it should work if you directly use a state here.

Still, it's great that you managed to implement a context 👍 .
Maybe it makes more sense when you read the avatar/user from your database?

Reply:
I want to use the avatar even more and integrate it on other sites, so this is already the basic framework for my next step.